src/components/room-link-validator.js

```diff
@@ -1,175 +0,0 @@
-/*
-Copyright 2020 The Matrix.org Foundation C.I.C.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-/**
- * The room link validator is used to determine if a room can be bridged.
- */
-const ConfigValidator = require("./config-validator");
-const log = require("./logging").get("room-link-validator");
-const VALIDATION_CACHE_LIFETIME = 30 * 60 * 1000;
-const PASSED = "RLV_PASSED";
-const ERROR = "RVL_ERROR";
-const ERROR_USER_CONFLICT = "RVL_USER_CONFLICT";
-const ERROR_CACHED = "RVL_ERROR_CACHED";
-
-const RULE_SCHEMA = {
-    "$schema": "http://json-schema.org/draft-04/schema#",
-    type: "object",
-    properties: {
-        userIds: {
-            type: "object",
-            properties: {
-                exempt: {
-                    type: "array",
-                    items: {
-                        type: "string"
-                    }
-                },
-                conflict: {
-                    type: "array",
-                    items: {
-                        type: "string"
-                    }
-                }
-            }
-        }
-    }
-};
-
-/**
- * The RoomLinkValidator checks if a room should be linked to a remote
- * channel, given a set of rules supplied in a config. The ruleset is maintained
- * in a seperate config from the bridge config. It can be reloaded by triggering
- * an endpoint specified in the {@link Bridge} class.
- */
-class RoomLinkValidator {
-    /**
-     * @param {string} config Config for the validator.
-     * @param {string} config.ruleFile Filename for the rule file.
-     * @param {string} config.rules Rules if not using a rule file, will be
-     *                              overwritten if both is set.
-     * @param {AppServiceBot} asBot The AS bot.
-     */
-    constructor (config, asBot) {
-        this.conflictCache = new Map(); // roomId => number
-        this.waitingRooms = new Map(); // roomId => Promise
```

Comment:
I guess this was meant to stop the same room from getting processed twice at the same time, but it wasn't implemented.